src/system.jl

```diff
@@ -0,0 +1,302 @@
+    zero_bp, one_bp, two_bp = String[], String[], String[]
+    for (name, breaks, mon) in zip(branches.name, branches.break_points, branches.is_monitored)
+        if mon
+            if length(breaks) == 0
+                push!(zero_bp, name)
+            elseif length(breaks) == 1
+                push!(one_bp, name)
+            else
+                push!(two_bp, name)
+            end
+        end
+    end
+    return zero_bp, one_bp, two_bp
+end
+
+"""
+    System
+
+The abstract type for representing the whole power system including topology, static
+components and their attributes, and time series data.
+
+Topology: `Dict`s linking generators, loads, and bids (if present) to buses.
+System wide static components and grid matrices: zones, buses, generators, branches, LODF and PTDF.
+Time series data: all the time series associated with generators, loads and bids.  All stored
+as `KeyedArray`s of `ids x datetimes`.
+"""
+abstract type System end
+
+"""
+    struct SystemDA <: System
+
+Subtype of a `System` for modelling the day-ahead market.
+
+Fields:
+ - Topology
+     - `gens_per_bus::Dict{String, Vector{Int}}`
+     - `incs_per_bus::Dict{String, Vector{String}}`
+     - `decs_per_bus::Dict{String, Vector{String}}`
+     - `psds_per_bus::Dict{String, Vector{String}}`
+     - `loads_per_bus::Dict{String, Vector{String}}`
+ - Static components
+     - `zones::Vector{Zone}`
+     - `buses::Buses`
+     - `generators::Generators`
+     - `branches::Branches`
+     - `LODF::Dict{String, KeyedArray}`
+     - `PTDF::KeyedArray`
+ - Time series
+     - `offer_curve::KeyedArray{Vector{Tuple{Float64, Float64}}}`
+     - `availability::KeyedArray{Bool}`
+     - `must_run::KeyedArray{Bool}`
+     - `regulation_min::KeyedArray{Float64}`
+     - `regulation_max::KeyedArray{Float64}`
+     - `pmin::KeyedArray{Float64}`
+     - `pmax::KeyedArray{Float64}`
+     - `ancillary_services::ServicesTimeSeries`
+     - `loads::KeyedArray{Float64}`
+     - `increment_bids::KeyedArray{Vector{Tuple{Float64, Float64}}}`
+     - `decrement_bids::KeyedArray{Vector{Tuple{Float64, Float64}}}`
+     - `price_sensitive_demand::KeyedArray{Vector{Tuple{Float64, Float64}}}`
+"""
+struct SystemDA <: System
+    gens_per_bus::Dict{String, Vector{Int}}
+    incs_per_bus::Dict{String, Vector{String}}
+    decs_per_bus::Dict{String, Vector{String}}
+    psds_per_bus::Dict{String, Vector{String}}
+    loads_per_bus::Dict{String, Vector{String}}
+
+    zones::Vector{Zone}
+    buses::Buses
+    generators::Generators
+    branches::Branches
+    LODF::Dict{String, KeyedArray}
+    PTDF::KeyedArray
+
+    # Generator related time series
+    offer_curve::KeyedArray{Vector{Tuple{Float64, Float64}}}
+    availability::KeyedArray{Bool}
+    must_run::KeyedArray{Bool}
+    regulation_min::KeyedArray{Float64}
+    regulation_max::KeyedArray{Float64}
+    pmin::KeyedArray{Float64}
+    pmax::KeyedArray{Float64}
+    ancillary_services::ServicesTimeSeries
```

Comment:
To be honest I'm still a bit unclear on the requirements for services.  I think this is because sometimes we won't always have the same set of four ancillary services (specifically there could be fewer I believe).  Raphael suggested a `Dict` for flexibility.  Not sure why I defined a type but the key is the services should be separate from the other time series data because they might or might not all be present in a `System`.